cmd/kubelet/app/server.go

```diff
@@ -355,13 +243,20 @@ func makeEventRecorder(s *componentconfig.KubeletConfiguration, kubeDeps *kubele
 }
 
 func run(s *options.KubeletServer, kubeDeps *kubelet.KubeletDeps) (err error) {
-	// TODO: this should be replaced by a --standalone flag
-	standaloneMode := (len(s.APIServerList) == 0 && !s.RequireKubeConfig)
+	// validate the initial KubeletServer
+	if err := validateKubeletServer(s); err != nil {
```

Comment:
this reordered validation to come before setting feature gates... seems like one of the things we'd want to do in validation is to validate options based on feature gates